### PR TITLE
Sites Management Page: Include `?v=<rounded-updated-at>` to bust mShot

### PR DESCRIPTION
**Pull request by danielbachhuber**
See conversation in https://github.com/Automattic/wp-calypso/issues/66524#issuecomment-1213372612

## Proposed Changes

Adds a `?v=<site-updated-at>` (rounded down to the nearest hour) to the `mShotsUrl` to semi-regularly refresh the screenshot when a site is updated.

I'm not sure if this is a "blessed" use of mShots. Notably, if the site's full page cache doesn't discard the `v` query param, this approach will likely bypass the full page cache, and take a longer time to generate a mShot.

## Testing Instructions

1. Navigate to `/sites-dashboard`.
2. Verify the `v` query parameter on a mShots URL is rounded down to the nearest start of the hour (e.g. `https://s0.wp.com/mshots/v1/https%3A%2F%2Fdanielwoatest.wpcomstaging.com%3Fv%3D2022-08-12T17%253A00%253A00.000Z?vpw=1200&vph=1200&w=748&countToRefresh=67`).
3. Comment out the `updatedAt.setMinutes( 0 );` and `updatedAt.setSeconds( 0 );` calls in this pull request.
4. Make a change to one of your sites.
5. Navigate back to the Sites Management Page and see the screenshot refresh once your sites data is refreshed.

### CODE REVIEW

File: client/sites-dashboard/components/sites-site-item-thumbnail.tsx [hunk 1]
```diff
@@ -22,10 +23,18 @@ export const SiteItemThumbnail = ( { site, ...props }: SiteItemThumbnailProps )
 
 	const shouldUseScreenshot = status === 'public';
 
+	let siteUrl = site.URL;
+	if ( site.options?.updated_at ) {
+		const updatedAt = new Date( site.options.updated_at );
+		updatedAt.setMinutes( 0 );
+		updatedAt.setSeconds( 0 );
+		siteUrl = addQueryArgs( siteUrl, { v: updatedAt.toISOString() } );
```
Comment (p-jackson) [inline client/sites-dashboard/components/sites-site-item-thumbnail.tsx]: Unix time would do just as well and is shorter and doesn't need as much escaping (i.e. no colons)

```suggestion
		siteUrl = addQueryArgs( siteUrl, { v: updatedAt.getTime() / 1000 } );
```

**Review by p-jackson — COMMENTED** [review]
The code looks good, but when I comment out the code that rounds to the nearest hour I don't see the SMP update 🤔 
Not sure what's going wrong.

**Comment by danielbachhuber** [comment]
> The code looks good, but when I comment out the code that rounds to the nearest hour I don't see the SMP update 🤔
> Not sure what's going wrong.

Do you see the timestamp in the URL? Does the timestamp change? What additional debugging have you done?

**Review by vykes-mac — COMMENTED** [review]
Tested it and the mshots was updated to reflect the changes done using the site editor.

**Review by sejas — APPROVED** [review]
Looking great!

- It's working great on simple sites.
- My screenshots for atomic sites are not getting updated because the `updated_at` is not updated immediately. I imagine that updated value gets updated by a `cron` for WoA. Nothing that we can fix in this PR.
- I think it's a smart a solid solution to update the thumbnails, as discussed here https://github.com/Automattic/wp-calypso/issues/66524
- It's great that the screenshots are updated automatically without needing to refresh the SMP.

**Pull request merged**